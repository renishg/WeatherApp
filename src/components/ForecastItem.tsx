import React, {FC} from 'react';
import {Image} from 'react-native';
import {View, Text, StyleSheet} from 'react-native';
import {colors} from '../theme/colors';
import { fonts } from '../theme/fonts';

type ForecastItemProps = {
  title: string;
  subTitle?: string;
  icon: string;
  tempC: number;
};

const ForecastItem: FC<ForecastItemProps> = ({
  title,
  subTitle,
  icon,
  tempC,
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>{title}</Text>
      {subTitle && <Text style={styles.subTitleText}>{subTitle}</Text>}
      <Image source={{uri: `https:${icon}`}} style={styles.icon} />
      <Text style={styles.tempText}>{`${tempC}Â°C`}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingHorizontal: 14,
  },
  titleText: {
    color: colors.white,
    fontSize: 16,
    fontFamily: fonts.semiBold,
  },
  subTitleText: {
    color: colors.white,
    fontSize: 12,
    fontFamily: fonts.regular,
  },
  icon: {
    width: 50,
    height: 50,
    marginVertical: 4,
  },
  tempText: {
    color: colors.white,
    fontSize: 16,
    fontFamily: fonts.semiBold,
  },
});

export {ForecastItem};
export type {ForecastItemProps};
