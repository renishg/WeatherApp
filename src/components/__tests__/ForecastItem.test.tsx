import React from 'react';
import renderer from 'react-test-renderer';
import {ForecastItem} from '../ForecastItem';

describe('ForecastItem', () => {
  it('renders correctly with subTitle', () => {
    const props = {
      title: 'Test-Title',
      subTitle: 'Test-SubTitle',
      icon: '//cdn.weatherapi.com/weather/64x64/night/296.png',
      tempC: 8,
    };
    const tree = renderer.create(<ForecastItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without subTitle', () => {
    const props = {
      title: 'Test-Title',
      icon: '//cdn.weatherapi.com/weather/64x64/night/296.png',
      tempC: 8,
    };
    const tree = renderer.create(<ForecastItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
