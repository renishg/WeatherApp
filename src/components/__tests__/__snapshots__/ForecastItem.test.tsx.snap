// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem renders correctly with subTitle 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "paddingHorizontal": 14,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 16,
      }
    }
  >
    Test-Title
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 12,
      }
    }
  >
    Test-SubTitle
  </Text>
  <Image
    source={
      {
        "uri": "https://cdn.weatherapi.com/weather/64x64/night/296.png",
      }
    }
    style={
      {
        "height": 50,
        "marginVertical": 4,
        "width": 50,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 16,
      }
    }
  >
    8°C
  </Text>
</View>
`;

exports[`ForecastItem renders correctly without subTitle 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "paddingHorizontal": 14,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 16,
      }
    }
  >
    Test-Title
  </Text>
  <Image
    source={
      {
        "uri": "https://cdn.weatherapi.com/weather/64x64/night/296.png",
      }
    }
    style={
      {
        "height": 50,
        "marginVertical": 4,
        "width": 50,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 16,
      }
    }
  >
    8°C
  </Text>
</View>
`;
